dace/sdfg/sdfg.py

```diff
@@ -1769,12 +1769,16 @@ def compile(self, output_file=None) -> \
         # Fill in scope entry/exit connectors
         sdfg.fill_scope_connectors()
 
+        # Serialize SDFG before generating code, since the codegen is allowed
+        # to mutate the SDFG internally
+        sdfg_json = sdfg.to_json(hash=True)
```

Comment:
That is definitely not something we want. Since the SDFG with the materialized default storage/schedules and line information is created during code generation, that's the one we want to save.

Reply:
The reason why I implemented this is exactly the opposite: as we discussed above, I would think that the user expects to see the same SDFG that they edited and saved, not one with mysterious changes done by the codegen!

That's why we are changing this in the first place: otherwise `program.sdfg` will be the version where the map has been unrolled into the SDFG, which is probably not what the user expects to see just because they put the unroll schedule on a map.